internal/graph/graph.go

```diff
@@ -178,18 +178,6 @@ func (g *Graph) recursiveDetectCycles(n graphNode, l []string) error {
 	return nil
 }
 
-func (g *Graph) validateGraph(ct reflect.Type) (reflect.Value, error) {
```

Comment:
We can talk about this behavior in person, but I don't think we should make everything zero value - if absent. I was going for this behavior - 

1. `Provide, Resolve` happens  on objects or Provided constructor that always return objects in a restrictive environment. The graph needs to be complete in order to resolve anything and zero value means we loose any context on why the graph is incomplete. I strongly feel we should error out on any incompleteness of the graph during resolve. This way, we maintain graph's completeness and know during build time that all objects "can be resolved" when graph is built.
2. `Invoke` is somewhat impromptu execution of methods that are not part of the graph and needs to pull parameters from dig. Since its an impromptu invocation, `dig` graph can't control graph completeness and should resolve missing dependencies in real time (using zero values). 

Making everything zero value effectively allowing dig to not have control over graph's completeness. We will lead to dig container into unstable states very often (think just executing without Providing anything to dig), and I am not very sure we are ready for it.